core/src/messaging/net_message.rs

```diff
@@ -132,6 +149,16 @@ impl NetMessage {
         &self.sender
     }
 
+    /// Returns the session of the `NetMessage`
+    pub fn session(&self) -> Option<SessionId> {
+        self.session
+    }
+
+    /// Sets the SessionId of the `NetMessage`
+    pub fn set_session(&mut self, session: SessionId) -> () {
```

Comment:
I think this function should be hidden from the user. Perhaps use pub(crate) or whatever that is needed to be used internally.